src/Aspire.Hosting/Dcp/DcpHostService.cs

```diff
@@ -62,7 +62,7 @@ internal sealed partial class DcpHostService : IHostedLifecycleService, IAsyncDi
             {
                 serviceCollection.AddSingleton(_applicationModel);
                 serviceCollection.AddSingleton(kubernetesService);
-                serviceCollection.AddScoped<IDashboardViewModelService, DashboardViewModelService>();
+                serviceCollection.AddSingleton<IDashboardViewModelService, DashboardViewModelService>();
```

Comment:
Why is `DashboardViewModelService` registered with a scope lifetime? I changed it because I wanted to use the dashboard as a dependency in another singleton service, and scope lifetime prevents that.

As far as I can tell, nothing has broken after changing the lifetime to singleton.

Reply:
I believe this was done as a way to clear the resources cache without the user having to do anything to do so, but still let the cache have some use (I believe in Blazor the scoped lifetime lasts across navigations as long as the page doesn't need to reload). Changing it to singleton would mean the cache is never cleared. @smitpatel can probably comment more.

Reply:
Ok. I changed both registrations to be scoped.

A downside to scoped registration is refreshing the page requires values to be added back into the cache, so refreshing on the trace view causes IP addresses to be displayed for a split second until the cache is populated and the page re-renders.